File: 2026-bogota-agent-sdk/.claude/skills/crear-agente-sdk/assets/entorno.py
"""entorno.py — carga la API key desde .env antes de que arranque el SDK.

Busca `.env` en la raíz del kit y en esta carpeta. Con `ANTHROPIC_API_KEY` el
agente corre contra tu cuenta de Console (así va en producción). Sin key, el SDK
usa la sesión de Claude Code de tu máquina.

    import entorno   # primera línea de cada script
"""

import os
from pathlib import Path

AQUI = Path(__file__).parent.resolve()

for ruta in (AQUI.parent / ".env", AQUI / ".env"):
    if ruta.exists():
        for linea in ruta.read_text().splitlines():
            linea = linea.strip()
            if linea and not linea.startswith("#") and "=" in linea:
                clave, valor = linea.split("=", 1)
                if valor.strip():
                    os.environ.setdefault(clave.strip(), valor.strip().strip("\"'"))

CON_KEY = bool(os.environ.get("ANTHROPIC_API_KEY"))
print(f"[auth] {'API key de .env (Console)' if CON_KEY else 'sin API key en .env: uso la sesión de Claude Code'}\n")

File: 2026-bogota-agent-sdk/.claude/skills/crear-agente-sdk/references/ejemplos/typescript/entorno.ts
// entorno.ts — carga la API key desde .env antes de que arranque el SDK.
//
// Busca `.env` en la raíz del kit y en esta carpeta. Con ANTHROPIC_API_KEY el agente corre
// contra tu cuenta de Console (así va en producción). Sin key, el SDK usa la sesión de
// Claude Code de tu máquina.
//
//   import "./entorno.ts";   // primera línea de cada script
import { existsSync, readFileSync } from "node:fs";
import { join } from "node:path";

for (const ruta of [join(import.meta.dirname, "..", ".env"), join(import.meta.dirname, ".env")]) {
  if (!existsSync(ruta)) continue;
  for (const cruda of readFileSync(ruta, "utf8").split("\n")) {
    const linea = cruda.trim();
    if (!linea || linea.startsWith("#") || !linea.includes("=")) continue;
    const i = linea.indexOf("=");
    const clave = linea.slice(0, i).trim();
    const valor = linea.slice(i + 1).trim().replace(/^["']|["']$/g, "");
    if (valor && !process.env[clave]) process.env[clave] = valor;
  }
}

export const CON_KEY = Boolean(process.env.ANTHROPIC_API_KEY);
console.log(`[auth] ${CON_KEY ? "API key de .env (Console)" : "sin API key en .env: uso la sesión de Claude Code"}\n`);
